app/Services/Meeting/MeetingService.php

```diff
@@ -32,6 +32,45 @@ public function create(Facility $facility, array $data): string
         return $meeting->id;
     }
 
+    public function patch(Meeting $meeting, array $data): void
+    {
+        $meeting->fill($data);
+
+        $newStaff = $this->extractStaff($data);
+        $newClients = $this->extractClients($data);
+
+        $currentStaff = $meeting->getAttendants(AttendanceType::Staff);
+        $currentClients = $meeting->getAttendants(AttendanceType::Client);
+
+        $finalStaff = empty($newStaff) ? $currentStaff : $newStaff;
+        $finalClients = empty($newClients) ? $currentClients : $newClients;
+
+        $finalAttendants = !empty($newStaff) || !empty($newClients) ? array_merge($finalStaff, $finalClients) : null;
+
+        // Resources is the only array that can be set to empty, so we need to distinguish that with null.
+        $finalResources = $this->extractResources($data, valueWhenAbsent: null);
```

Comment:
walidator jest taki:
```php
'staff', 'clients', 'resources' => Valid::list(sometimes: true, min: 0),
```
więc każda z tych list może być pusta, a zdaje się, że nie jest co całkiem obsłużone
wydaje mi się, że warto zmienić metody extract* z `array` na `?array`, i gdy pole nie przyszło, zwracać null

poza tym ja unikam `empty`, bo ono wyłącza sprawdzanie sprawdzanie zadeklarowania zmiennych w ramach statycznej analizy kodu, `count($a)===0`, lub samo `$a` (pusta tablica jest false)

Reply:
Zrobione, chociaż uważam, że przynajmniej staff nie powinno móc być puste - UI kalendarza wymaga wybrania dokładnie jednego pracownika, żeby wyświetlić jego spotkania.

Ale to można również osobno dopisać.